frontend/__tests__/actions/confirm-phone-number.test.tsx

```diff
@@ -4,20 +4,34 @@ import { describe, expect, it, vi } from 'vitest';
 
 import { action } from '~/routes/_gcweb-app.personal-information.phone-number.confirm';
 
+vi.mock('~/services/user-service.server.ts', () => ({
+  userService: {
+    getUserId: vi.fn().mockReturnValue('00000000-0000-0000-0000-000000000000'),
+    getUserInfo: vi.fn().mockReturnValue({
+      firstName: 'John',
+      homeAddress: '123 Home Street',
+      lastName: 'Maverick',
+      mailingAddress: '123 Mailing Street',
+      phoneNumber: '(555) 555-5555',
+      preferredLanguage: 'fr',
+    }),
+  },
+}));
+
 describe('Confirm Phone Number Action', () => {
   afterEach(() => {
     vi.clearAllMocks();
     vi.resetModules();
   });
 
-  it('Should redirect', async () => {
+  it('Should redirect to homepage page when newPhoneNumber is missing', async () => {
     let request = new Request('http://localhost:3000/personal-information/phone-number/confirm', {
       method: 'POST',
     });
 
     const response = await action({ request, context: {}, params: {} });
 
     expect(response.status).toBe(302);
-    expect(response).toEqual(redirect('/personal-information/phone-number/success'));
+    expect(response).toEqual(redirect('/'));
```

Comment:
If the expected is `/`, wouldn't that imply that action never reaches the final return (as in there's something likely missing in the mocks you have).

Reply:
I'll add in the missing test which requires mocking of the session and push a commit to this PR shortly

Reply:
ehh I just now realized your test is correct based on the `it` block statement. Whoops.  Honestly I don't even really care if you put it in or not.  We'll definitely be testing out this flow for e2e tests though.

Reply:
Added in missing mock test as per discussion